### PR TITLE
Group some sections of our logs in github actions

### PR DESCRIPTION
**Pull request by oli-obk**
This makes logs a little bit more readable as you can now collapse all the parts that don't interest you (and they get collapsed automatically)

Obviously there's a lot more sites where we can/need to do this, too, but this is already helpful imo

r? @jyn514

### CODE REVIEW

**Comment by oli-obk** [comment]
Here's how it looks on my phone:

Screenshot_Fennec_20230421-085430~2

**Review by jyn514 — COMMENTED** [review]
This is really awesome! I didn't realize this was even possible. I think it's going to make https://github.com/rust-lang/rust/issues/97338 a lot easier too 😍 doing this as a function on `builder` instead of part of the `Step` was a good idea, reduces boilerplate.

I want to go over the CI logs and make sure there's no regressions (I think I saw a few extra spaces which are NBD but would be nice to fix), but I'm out of time for today - hopefully this is useful in the meantime.

File: src/bootstrap/lib.rs [hunk 1]
```diff
@@ -181,6 +182,30 @@ pub enum GitRepo {
...
+    /// not really work.
+    pub fn group(name: impl std::fmt::Display) -> Group {
+        if std::env::var_os("GITHUB_ACTIONS").is_some() {
+            eprintln!("::group::{name}");
+        }
+        Group(())
+    }
+
+    /// A guard that closes the current github actions log group on drop.
+    #[must_use]
```
[the remark this answers is not in the document]
Comment (oli-obk) [reply src/bootstrap/lib.rs:197]: No, it just warns for `group();`

File: src/bootstrap/lib.rs [hunk 2]
```diff
@@ -993,6 +1018,61 @@ impl Build {
...
+            if host == target {
+                msg(format_args!(" ({target})"))
+            } else {
+                msg(format_args!(" ({host} -> {target})"))
+            }
+        } else {
+            msg(format_args!(""))
+        };
+        self.info(&msg);
+        gha::group(&msg)
```
[the remark this answers is not in the document]
Comment (oli-obk) [reply src/bootstrap/lib.rs:1042]: Ah yea, I've seen those

File: src/bootstrap/lib.rs [hunk 3]
```diff
@@ -993,6 +994,85 @@ impl Build {
...
+                msg(format_args!(" ({host} -> {target})"))
+            }
+        } else {
+            msg(format_args!(""))
+        };
+        self.group(&msg)
+    }
+
+    /// Return a `Group` guard for a [`Step`] that is only built once and isn't affected by `--stage`.
+    fn msg_sysroot_tool(
```
Comment (jyn514) [inline src/bootstrap/lib.rs:1039]: I think this new name should go on msg_staged, not what used to be msg_unstaged. msg_unstaged was fine.

**Review by jyn514 — APPROVED** [review]
Thanks! I want to add lots more groups but we don't need to block this PR on that, it's already a big improvement :)

r=me with the name change

**Pull request merged**